File: src/main/java/net/cortexmodders/atomtech/item/circuits/DataConversion.java
package net.cortexmodders.atomtech.item.circuits;

public class DataConversion
{
	public static long getBytes(long bits)
	{
		return bits / 8;
	}
	
	public static long getKiloBytes(long bits)
	{
		return getBytes(bits) / 1024;
	}
	
	public static long getMegaBytes(long bits)
	{
		return getKiloBytes(bits) / 1024;
	}
	
	public static long getGigaBytes(long bits)
	{
		return getMegaBytes(bits) / 1024;
	}
	
	public static long getTeraBytes(long bits)
	{
		return getGigaBytes(bits) / 1024;
	}
	
	public static long getPetaBytes(long bits)
	{
		return getTeraBytes(bits) / 1024;
	}
	
	public static long getExaBytes(long bits)
	{
		return getPetaBytes(bits) / 1024;
	}
	
	public static long getZettaBytes(long bits)
	{
		return getExaBytes(bits) / 1024;
	}
	
	public static long getYottaBytes(long bits)
	{
		return getZettaBytes(bits) / 1024;
	}
}

File: src/main/scala/net/cortexmodders/atomtech/item/circuits/DataConversion.scala
package net.cortexmodders.atomtech.item.circuits

class DataConversion
{
	def getBytes(bits: Double) = bits / 8.0D
	
	def getKiloBytes(bits: Double) = getBytes(bits) / 1024.0D
	
	def getMegaBytes(bits: Double) = getKiloBytes(bits) / 1024.0D
	
	def getGigaBytes(bits: Double) = getMegaBytes(bits) / 1024.0D
	
	def getTeraBytes(bits: Double) = getGigaBytes(bits) / 1024.0D
	
	def getPetaBytes(bits: Double) = getTeraBytes(bits) / 1024.0D
	
	def getExaBytes(bits: Double) = getPetaBytes(bits) / 1024.0D
	
	def getZettaBytes(bits: Double) = getExaBytes(bits) / 1024.0D
	
	def getYottaBytes(bits: Double) = getZettaBytes(bits) / 1024.0D
}